Lib/asyncio/base_events.py

```diff
@@ -458,16 +458,26 @@ def create_future(self):
         """Create a Future object attached to the loop."""
         return futures.Future(loop=self)
 
-    def create_task(self, coro, **kwargs):
+    def create_task(self, coro, *, name=None, context=None, **kwargs):
         """Schedule a coroutine object.
 
         Return a task object.
         """
         self._check_closed()
         if self._task_factory is not None:
-            return self._task_factory(self, coro, **kwargs)
+            if context is not None:
+                kwargs["context"] = context
 
-        task = tasks.Task(coro, loop=self, **kwargs)
+            task = self._task_factory(self, coro, **kwargs)
+            task.set_name(name)
+            try:
+                # gh-128552: prevent a refcycle of
+                # task.exception().__traceback__->BaseEventLoop.create_task->task
+                return task
+            finally:
+                del task
```

Comment:
Can we refactor so that this try/except appears around the entire function body rather than being duplicated twice?